src/Avalonia.OpenGL/Composition/CompositionGlContextExtensions.cs

```diff
@@ -0,0 +1,45 @@
+using System;
+using Avalonia.Rendering.Composition;
+
+namespace Avalonia.OpenGL.Composition;
+
+public static class CompositionGlContextExtensions
+{
+    public static ICompositionGlSwapchain CreateSwapchain(this ICompositionGlContext context, Visual visual, PixelSize size)
```

Comment:
I would expect this method to have API semantics of `CreateSurfaceVisual` and `SetElementChildVisual(visual, surfaceVisual)` combined. As it's not much clear from the method name, that compositor visual is involved at all, without looking in the implementation.

Maybe something like `CreateAndSetVisualSwapchain`? Some docs would help as well.

Reply:
somewhat not agree with you Max, because swapchain is not for Visual. it's a standalone object for GPU buffer presenting, it's part of OpenGl although it's optional in a rendering system.